GUIDANCE.md

```diff
@@ -0,0 +1,246 @@
+
+```ruby
+customer_lines = customers_file.split("\n")
+```
+
+We tell ruby to `.split` the string using the delimiter `\n`. After that we have an Array with String elements. Now go and print the 13th (or any other) element from that array:
+
+```ruby
+puts customer_lines[12]
+# => 13;Shae;Gildersleeve;Mumbai
+```
+
+#### Splitting a file's row into cells
+
+Next, we're going to split **each line** into "cells".
+
+```ruby
+customer_cells = customer_lines[12].split(";")
+# => ["13", "Shae", "Gildersleeve", "Mumbai"]
+```
+
+Once again we're using `.split` to separate that String into multiple elements.
+Finally, we can assign the individual elements to meaningful variables:
+
+```ruby
+customer_id = customer_cells[0]
+first_name = customer_cells[1]
+last_name = customer_cells[2]
+location = customer_cells[3]
+```
+
+With this setup, we can start using the variables as usual. For example we could print customer info like this:
+
+```ruby
+puts "Customer #{id}: #{first_name} #{last_name} (#{location})"
+```
+
+## Excercise
+
+### Overview
+
+This exercise consists of 3 Ruby files and 2 CSV files.
+
+The Ruby files are
+
+| Filename            | Comment                                     |
+|---------------------|---------------------------------------------|
+| `order_analysis.rb` | This class has to be modified and extended. |
+| `customer.rb`       | This class has to be implemented entirely.  |
+| `order.rb`          | This class has to be implemented entirely.  |
+
+The CSV files are
+
+| Filename       | Comment                                               |
+|----------------|-------------------------------------------------------|
+| `customer.csv` | Contains customer data.                               |
+| `order.csv`    | Contains order data, with reference to customer data. |
+
+### Running the script
+
+We can execute `order_analysis.rb`. Please execute it and investigate the output.
+
+```shell
+ruby order_analysis.rb
+```
+
+Then have a look at `order_analysis.rb` and _try to understand_ what's going on in the existing code.
+
+### Milestones
+
+Let's dive into the actual exercise. We split it into several milestones, which should make it more digestible. :wink:
+
+Please note that the following approach is just _one of many_. It is NOT the only way how to work with CSV-input data.
+
+#### Milestone 1: The `Customer` class
+
+The `Customer` class (or "model") will be defined in `customer.rb`.
+
+1. Create a **constructor** that takes all customer data (`id`, `first_name`, etc.) as individual input arguments.
+2. Assign each input argument to **instance variables** and define corresponding _getter methods_.
+3. Then define a `print_info` method, that prints a nicely formatted customer information, for example `"Customer 13: Shae Gildersleeve (Zurich)"`
+4. Finally, use the `Customer` class in `OrderAnalysis`, and remove obsolete lines.
+
+After these steps, the result should be the **same** as before:
```

Comment:
But the initial code in `order_analysis.rb` doesn't print the location? Maybe we should add that there?

Reply:
Yes, very good find. I added it to the code.